apps/src/lib/tools/jsdebugger/JsDebugger.jsx

```diff
@@ -205,7 +233,119 @@ var UnconnectedJsDebugger = Radium(React.createClass({
+    var offset = (parseInt(window.getComputedStyle(codeApp).bottom, 10) || 0) -
+                 rect.height / 2;
+    var newDbgHeight = Math.max(
+      MIN_DEBUG_AREA_HEIGHT,
+      Math.min(
+        MAX_DEBUG_AREA_HEIGHT,
+        (window.innerHeight - event.pageY) - offset
+      )
+    );
+
+    if (!this.isOpen()) {
+      this.slideOpen();
+    }
+
+    codeTextbox.style.bottom = newDbgHeight + 'px';
+    this.root.style.height = newDbgHeight + 'px';
+
+    // Fire resize so blockly and droplet handle this type of resize properly:
+    utils.fireResizeEvent();
+  },
+
+  onMouseDownWatchersResizeBar(event) {
+    // When we see a mouse down in the resize bar, start tracking mouse moves:
+    var eventSourceElm = event.srcElement || event.target;
+    if (eventSourceElm.id === 'watchersResizeBar') {
+      this._draggingWatchersResizeBar = true;
+      document.body.addEventListener('mousemove', this.onMouseMoveWatchersResizeBar);
+      const mouseMoveTouchEventName = dom.getTouchEventName('mousemove');
+      if (mouseMoveTouchEventName) {
+        document.body.addEventListener(
+          mouseMoveTouchEventName,
+          this.onMouseMoveWatchersResizeBar
+        );
+      }
+
+      event.preventDefault();
+    }
+  },
+
+  onMouseUpWatchersResizeBar() {
+    // If we have been tracking mouse moves, remove the handler now:
+    if (this._draggingWatchersResizeBar) {
+      document.body.removeEventListener('mousemove', this.onMouseMoveWatchersResizeBar);
+      const mouseMoveTouchEventName = dom.getTouchEventName('mousemove');
+      if (mouseMoveTouchEventName) {
+        document.body.removeEventListener(
+          mouseMoveTouchEventName,
+          this.onMouseMoveWatchersResizeBar
+        );
+      }
+      this._draggingWatchersResizeBar = false;
+    }
+  },
+
+  /**
+   *  Handle mouse moves while dragging the debug resize bar.
+   */
+  onMouseMoveWatchersResizeBar(event) {
+    const watchers = this._watchers.getWrappedInstance();
+    const watchersRect = watchers.scrollableContainer.getBoundingClientRect();
+    const movement = watchersRect.left - event.clientX;
+    const newDesiredWidth = watchersRect.width + movement;
+    const newWatchersWidth = Math.max(
+      MIN_WATCHERS_AREA_WIDTH,
+      Math.min(MAX_WATCHERS_AREA_WIDTH, newDesiredWidth)
+    );
+
+    const watchersResizeRect = this._watchersResizeBar.getBoundingClientRect();
+    const watchersResizeRight = (newWatchersWidth - watchersResizeRect.width / 2);
+
+    watchers.scrollableContainer.style.width = newWatchersWidth + 'px';
+    this._debugConsole.getWrappedInstance().root.style.right = newWatchersWidth + 'px';
+    this._watchersResizeBar.style.right = watchersResizeRight + 'px';
+
+    const headerLBorderWidth = 1;
+    const watchersLRBorderWidth = 2;
+    const extraWidthForHeader = watchersLRBorderWidth - headerLBorderWidth;
+    this._debugWatchHeader.root.style.width = newWatchersWidth + extraWidthForHeader + 'px';
+  },
```

Comment:
all this code for handling draggable split panes should really be broken out into a standalone ui component. There are some react libraries that do this (like https://github.com/tomkp/react-split-pane), but last I checked they were a bit buggy. Another change for another day.